### ISSUE TITLE
[BREAKING] Rename `model`'s Python modules and expose `_base`

### ISSUE THREAD

**Issue by sdatkinson** [sdatkinson/neural-amp-modeler#508]
There's confusing naming. `nam.models.base` holds the PyTorch _Lightning Module_ that combines the mdoel with its optimizer, loss functions, etc; and `_base` which holds the base of the actual _model_.

It's also undesirable for `_base`, which is really helpful, to be private.

Changes:
* Move `nam.models.base` to `nam.train.lightning_module` to emphasize its relevance to _training_.
* Move `nam.models._base` to `nam.models.base`.

It's a bit confusing to put `_base` where `base` just was, but this is a nicer organization of the code IMO and better set up other code to (safely) build on NAM's abstractions.
